Sources/Sentry/SentryCrashScopeObserver.m

```diff
@@ -127,13 +127,19 @@ - (void)syncScope:(id)object
         return;
     }
 
-    id serialized = serialize();
+    id serialized;
+    @synchronized(self) {
```

Comment:
`h`: I think it would be better to solve the problem directly in SentryScope.m https://github.com/getsentry/sentry-cocoa/blob/e0abb7e49c3133b01a2ad0307415661289c2d99e/Sources/Sentry/SentryScope.m#L338-L345

similar to how we do it for other properties there https://github.com/getsentry/sentry-cocoa/blob/e0abb7e49c3133b01a2ad0307415661289c2d99e/Sources/Sentry/SentryScope.m#L317-L329

because then we solve the issue for all scope observers.

Another downside of this approach is that we now acquire two locks for every scope sync, even though all other properties are already thread-safe in Scope.m. As this gets called mostly on the main thread, acquiring as few locks as possible is essential.

Reply:
I checked this approach, and the issue really is in the serialization in the observers itself, not in scope. Synchronizing the user in the scope did not prevent this particular crash, but i changed it to just one sync.

<img width="2062" alt="Bildschirmfoto 2024-08-14 um 11 09 11" src="https://github.com/user-attachments/assets/c38d8a8e-3b3a-4225-a9dc-d0e4982922f1">

Reply:
If you change the setUser method in the scope to this
```
- (void)setUser:(SentryUser *_Nullable)user
{
    @synchronized(self) {
        
        self.userObject = user;
        
        for (id<SentryScopeObserver> observer in self.observers) {
            [observer setUser:user];
        }
    }
}
```

it works. If you synchronize the user like this it doesn't work, because synchronized is based on the object instance. See [docs](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Multithreading/ThreadSafety/ThreadSafety.html):

> The object passed to the @synchronized directive is a unique identifier used to distinguish the protected block. If you execute the preceding method in two different threads, passing a different object for the anObj parameter on each thread, each would take its lock and continue processing without being blocked by the other. If you pass the same object in both cases, however, one of the threads would acquire the lock first and the other would block until the first thread completed the critical section.

```
- (void)setUser:(SentryUser *_Nullable)user
{
    @synchronized(self.userObject) {
        
        self.userObject = user;
        
        for (id<SentryScopeObserver> observer in self.observers) {
            [observer setUser:user];
        }
    }
}
```

Reply:
Ah yes, missed this 🤦 thank you!

Reply:
This also happened to me several times until I remembered it. This cost me hours in the past 🤦 😄